profiling/variorum-connector/variorum-connector.cpp

```diff
@@ -226,27 +267,138 @@ void variorum_call_mpi() {
+    if (num_sockets_pos != nullptr) {
+      num_sockets_pos += strlen("\"num_gpus_per_socket\":");
+
+      char *num_sockets_end_pos = strchr(num_sockets_pos, ',');
+      if (num_sockets_end_pos == nullptr) {
+        num_sockets_end_pos = strchr(num_sockets_pos, '}');
+      }
+      if (num_sockets_end_pos != nullptr) {
+
+        std::string num_sockets_str(num_sockets_pos,
+                                    num_sockets_end_pos - num_sockets_pos);
+
+        num_sockets = std::stoll(num_sockets_str);
+        //          std::cout << "Number of Sockets: " << num_sockets <<
+        //          std::endl;
+      }
+    }
+    char *power_gpu_watts_pos = strstr(s, "\"power_gpu_watts\":");
+
+    if (power_gpu_watts_pos != nullptr) {
+      // for (int gpu_num = 0; gpu_num <= num_sockets; ++gpu_num) {
+      for (int gpu_num = 0; gpu_num <= gdevID; ++gpu_num) {
+        std::string gpu_key = "\"GPU_" + std::to_string(gpu_num) + "\":";
+        char *gpu_pos = strstr(power_gpu_watts_pos, gpu_key.c_str());
+        if (gpu_pos != nullptr) {
+          gpu_pos += gpu_key.length();
+          char *gpu_end_pos = strchr(gpu_pos, ',');
+          if (gpu_end_pos == nullptr) {
+            gpu_end_pos = strchr(gpu_pos, '}');
+          }
+          if (gpu_end_pos != nullptr) {
+            std::string gpu_power_str(gpu_pos, gpu_end_pos - gpu_pos);
+            float gpu_power = std::stof(gpu_power_str);
+            //      std::cout << "GPU_" << gpu_num << " Power: " << gpu_power <<
+            //      " Watts  " ;
+            if (power == 0) {
+              gpu_powers.push_back(gpu_power);
+            } else {
+              gpu_powers2.push_back(gpu_power);
+            }
+            if (power == -1) {
+              power1 += gpu_power;
+              //	std::cout << power1 <<" power currently \n" <<
+              //std::endl; 	std::cout << power << "   power \n" << std::endl;
+            } else {
+              power2 += gpu_power;
+              // std::cout << power << "   power \n" << std::endl;
+            }
+            // std::cout << power << "  power1 \n"<< std::endl;
+          }
+        }
+      }
+      power += 1;
+      if (power2 != 0) {
+        //	std::cout << "times" << time2 << "  "<< time1 << std::endl;
+
+        //	std::cout << "Energy Estimation\t" <<
+        //((power2+power1)/2)*((time2-time1)*.001) << std::endl;
+      }
+      // std::cout << "power 1 and 2 " << power1 << " " << power2;
+      if (power2 != 0 && power1 != 0) {
+        power1 = 0;
+        power2 = 0;
+        power = -1;
+        temp = time2;
+        time2 = 0;
+      }
+    }
+    if (power != 0) {
+      std::cout << "  Gpu start power: " << gpu_powers2[gdevID]
+                << " Gpu end power: " << gpu_powers[gdevID]
+                << "   Energy Estimation:   "
+                << ((gpu_powers[gdevID] + gpu_powers2[0]) / 2) *
+                       ((temp - time1) * .001)
+                << std::endl;
```

Comment:
Let's just not print anything to stdout but only to the file.

Reply:
Sounds fine to me.